internal/controller/dnsrecord_healthchecks.go

```diff
@@ -94,6 +95,69 @@ func (r *DNSRecordReconciler) ensureProbe(ctx context.Context, generated *v1alph
 	return nil
 }
 
+// removeUnhealthyEndpoints fetches all probes associated with this record and uses the following criteria while removing endpoints:
+//   - If the Leaf Address has no health check CR - it is healthy
+//   - If the health check CR has insufficient failures - it is healthy
+//   - If the health check CR is deleting - it is healthy
+//   - If the health check is a CNAME and any IP is healthy - the CNAME is healthy
+func (r *DNSRecordReconciler) removeUnhealthyEndpoints(ctx context.Context, endpoints []*endpoint.Endpoint, dnsRecord *v1alpha1.DNSRecord) ([]*endpoint.Endpoint, bool, error) {
```

Comment:
what is the boolean for?

Reply:
for reporting to status. Changed to string slice for more information in the next PR. I decided to add it here so the next PR will not be changing this file, but I ended up refactoring it anyway. Could remove for this one if this matters